src/options/smt_options.toml

```diff
@@ -220,6 +220,14 @@ header = "options/smt_options.h"
   notifies   = ["notifyBeforeSearch"]
   help       = "turn on unsat core generation"
 
+[[option]]
+  name       = "minimalUnsatCores"
```

Comment:
The unsat cores you are generating here are not guaranteed to be minimal, right? In that case I'd prefer `minimizeUnsatCores` rather than `minimalUnsatCores`.

Reply:
if it is not minimal, then I would prefer `smallUnsatCore` or `reducedUnsatCore`

Reply:
They are guaranteed to be _minimal_ unsat cores (i.e. removing any of the assertions makes the constraints satisfiable) but not _minimum_ unsat cores (following the terminology e.g. [here](https://en.wikipedia.org/wiki/Unsatisfiable_core) and [here](https://arxiv.org/ftp/arxiv/papers/1401/1401.3878.pdf)). The only caveat is that we may not end up with minimal unsat cores when we get `unknown` answers. What do you think?

Reply:
I think that the Wikipedia article you linked is nice. :) But the terminology is confusing regardless. The key difference is between local vs global, so it seems a bit arbitrary to use "minimal" for the local case but "minimum" for the global one.  I'd go with `smallUnsatCore` just to be non-committal.

Reply:
It is common to use `minimal` if it can't be further reduced but is not globally the minimum, but only locally. I personally don't like `smallUnsatCore` -- it doesn't imply what it actually is, minimal in the sense that if you remove any assertion it's going to be sat.